src/test/coqLsp/coqLspCommandInGoalsReq.test.ts

```diff
@@ -0,0 +1,239 @@
+            position,
+            fileUri,
+            1,
+            command
+        );
+        await client.closeTextDocument(fileUri);
+        client.dispose();
+
+        return goals;
+    }
+
+    test("One Coq sentence: fail to solve with bad goal", async () => {
+        const goal = await getGoalAtPosition(
+            { line: 9, character: 4 },
+            ["small_document.v"],
+            "reflexivity."
+        );
+
+        expect(goal.err).toEqual(true);
+        expect(goal.val).toBeA(Error);
+        if (goal.val instanceof Error) {
+            expect(goal.val.message).toEqual(
+                'In environment\nn : nat\nUnable to unify "n" with\n "0 + n + 0".'
+            );
+        }
+    });
+
+    test("One Coq sentence: fail to solve with bad tactic", async () => {
+        const goal = await getGoalAtPosition(
+            { line: 9, character: 4 },
+            ["small_document.v"],
+            "reflexivit."
+        );
+
+        expect(goal.err).toEqual(true);
+        expect(goal.val).toBeA(Error);
+        if (goal.err) {
+            expect(goal.val.message).toEqual(
+                "The reference reflexivit was not found in the current\nenvironment."
+            );
+        }
+    });
+
+    test("One Coq sentence: successfully solve no goals left", async () => {
+        const goals = await getGoalAtPosition(
+            { line: 9, character: 4 },
+            ["small_document.v"],
+            "auto."
+        );
+
+        expect(goals.ok).toEqual(true);
+        if (goals.ok) {
+            expect(goals.val).toBeEmpty();
+        }
+    });
+
+    test("One Coq sentence: successfully solve 1 goal left", async () => {
+        const goals = await getGoalAtPosition(
+            { line: 9, character: 4 },
+            ["test_many_subgoals.v"],
+            "auto."
+        );
+
+        expect(goals.ok).toEqual(true);
+        if (goals.ok) {
+            expect(goals.val).toHaveLength(1);
+            expect(goals.val[0].ty).toEqual("S n + 0 = S n");
+        }
+    });
+
+    test("One Coq sentence: successfully solve 2 goals left", async () => {
+        const goals = await getGoalAtPosition(
+            { line: 22, character: 4 },
+            ["test_many_subgoals.v"],
+            "auto."
+        );
+
+        expect(goals.ok).toEqual(true);
+        if (goals.ok) {
+            expect(goals.val).toHaveLength(2);
+            expect(goals.val[0].ty).toEqual(
+                "Second = First \\/ Second = Second \\/ Second = Third"
+            );
+            expect(goals.val[1].ty).toEqual(
+                "Third = First \\/ Third = Second \\/ Third = Third"
+            );
+        }
+    });
+
+    test("One Coq sentence wrapped into curly braces: solve successfully", async () => {
+        const goals = await getGoalAtPosition(
+            { line: 9, character: 4 },
+            ["small_document.v"],
+            " { auto. }"
+        );
+
+        expect(goals.ok).toEqual(true);
```

Comment:
This block of code also repeats at least two times... Maybe you will even manage to move most of the successful checks into a single pretty abstract method 🙏